### ISSUE TITLE
Support for polygon-based subsetting

### ISSUE THREAD

**Issue by jerstlouis** [opengeospatial/GeoDataCube-SWG#6]
From Testbed 19 GDC ER Critical Feedback:

> 4. OGC API - Coverages does not allow for polygon-based subsetting

That is possible with [OGC API - Coverages - Part 2: Filtering, deriving fields, aggregation and convolution](https://github.com/opengeospatial/ogcapi-coverages/issues/164), using a CQL2 filter with `S_INTERSECTS` and a `POLYGON` geometry, as described in the [ER in section 5.1.2.6.  Future work: geometry intersections, spatial joins, aggregation and convolution](http://ogc.pages.ogc.org/T19-GDC/documents/D011/document.html#_36196079-1d05-419c-98da-0828e02d2844):

`filter=S_INTERSECTS(rec.cells, POLYGON((-109 64.39894807, -61.25 64.39894807,-61.25 76.7652818, -109 76.7652818, -109 64.39894807)))`

The reason no draft document for [Part 2](https://github.com/opengeospatial/ogcapi-coverages/issues/164) exists yet is only because we are trying to focus on finalizing Part 1, but if it would be useful to raise awareness of these envisaged capabilities, maybe we should really create a draft document that can evolve in parallel.

NOTE: OGC API - EDR queries also support this use case. Most EDR queries are likely mappable to such an OGC API - Coverages request with a CQL2 filter.
